src/realm/query.cpp

```diff
@@ -1004,20 +998,18 @@ R Query::aggregate(ColKey column_key, size_t* resultcount, ObjKey* return_ndx) c
             for (size_t t = 0; t < m_view->size(); t++) {
                 const Obj obj = m_view->get_object(t);
                 if (eval_object(obj)) {
-                    st.template match<action, false>(size_t(obj.get_key().value), 0, obj.get<T>(column_key));
+                    st.match(size_t(obj.get_key().value), obj.get<T>(column_key));
```

Comment:
This cast to size_t for example is not safe. It's a preexisting problem, but it's kinda related to the clean up in this PR.

Reply:
Fixed. Perhaps not in the most elegant way - but anyway fixed.